src/core/middlewares/chart/builder.go

```diff
@@ -29,8 +29,8 @@ import (
 )
 
 var (
-	deleteChartVersionRe = regexp.MustCompile(`^/api/chartrepo/(?P<namespace>\w+)/charts/(?P<name>\w+)/(?P<version>[\w\d\.]+)/?$`)
-	createChartVersionRe = regexp.MustCompile(`^/api/chartrepo/(?P<namespace>\w+)/charts/?$`)
+	deleteChartVersionRe = regexp.MustCompile(`^/api/chartrepo/(?P<namespace>[^?#]+)/charts/(?P<name>[^?#]+)/(?P<version>[^?#]+)/?$`)
```

Comment:
can it support "123/123/123.tgz"? actually, I can give any name for the tgz when to load the chat file.

Reply:
I tested the chart deletion URL (`"/api/chartrepo/:repo/charts/:name/:version"`) in harbor, the `repo`, `name` and `version` do not match string include `/`.

So `"/api/chartrepo/library/charts/123/123/123/1.0"` will 404 in harbor.

The URL pattern `"/api/:repo/charts/:name/:version"` of `chartmuseum` also do not match string include `/` in `repo`, `name` and `version`.

Reply:
but, I can upload the `123/123/123.tgz` with harbor UI.